src/shared/utilities/vsCodeUtils.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+export const localize = nls.loadMessageBundle()
+
+import { ErrorOrString, getLogger, Logger, LogLevel } from '../logger'
+
+// ------- Experimental combined output channel & traditional logger  --------------
+
+interface LogParams {
+    args: ErrorOrString[],
+    channel: vscode.OutputChannel,
+    level: LogLevel,
+    logger: Logger,
+    msg: string,
+    nlsKey: string,
+}
+
+// TODO: Leverage 3rd party code: https://raw.githubusercontent.com/Microsoft/vscode-nls/master/src/main.ts
+// function format(template: string, ...templateParams: any[]): string {
```

Comment:
Please remove the signature from the comment until we add attribution.

Reply:
✔️